python_runner/snoop.py

```diff
@@ -0,0 +1,55 @@
+import ast
+import inspect
+import os
+import sys
+
+import snoop
+import snoop.formatting
+import snoop.tracer
+
+internal_dir = os.path.dirname(os.path.dirname(
+    (lambda: 0).__code__.co_filename
+))
+snoop.tracer.internal_directories += (
+    internal_dir,
+)
+
+
+def exec_snoop(runner, code_obj, out=sys.stdout, color=True):
+    class PatchedFrameInfo(snoop.tracer.FrameInfo):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.is_ipython_cell = self.frame.f_code == code_obj
+
+    snoop.tracer.FrameInfo = PatchedFrameInfo
+
+    snoop.formatting.Source._class_local('__source_cache', {}).pop(runner.filename, None)
+
+    config = snoop.Config(
+        columns=(),
+        out=out,
+        color=color,
+    )
+    tracer = config.snoop()
+    tracer.variable_whitelist = set()
+    for node in ast.walk(ast.parse(runner.source_code)):
+        if isinstance(node, ast.Name):
+            name = node.id
+            tracer.variable_whitelist.add(name)
+    tracer.target_codes.add(code_obj)
+
+    def find_code(root_code):
+        """
+        Trace all functions recursively, like trace_module_deep.
```

Comment:
I think trace_module_deep is only a thing in birdseye, this is the result of some copy pasting. This comment should probably be fixed in futurecoder as well.